### ISSUE TITLE
Bug(`no-duplicate-schema-methods`): allow duplicate `regex` calls with different values

### ISSUE THREAD

**Issue by noahbald** [marcalexiei/eslint-zod#434]
### 📦 Affected package

eslint-plugin-zod

### 🧩 Description

It's common to use multiple regex definitions for a schema value. Consider allowing.

### 🪜 How to reproduce

This produces an lint error.

> Method `.regex()` is called more than once in this schema chain

```ts
const ASCII_RANGE = /^[ -~]*$/;
const NON_LATIN = /^W/;
const PasswordSchema = z.string().regex(ASCII_RANGE).regex(NON_LATIN);
```

### ✅ Expected behavior

This should not produce an lint error.

```ts
const ASCII_RANGE = /^[ -~]*$/;
const NON_LATIN = /^W/;
const PasswordSchema = z.string().regex(ASCII_RANGE).regex(NON_LATIN);
```

However, these could still produce a lint error.

```ts
const ASCII_RANGE = /^[ -~]*$/;
const PasswordSchema = z.string().regex(ASCII_RANGE).regex(ASCII_RANGE);
```
```ts
const PasswordSchema = z.string().regex(/^[ -~]*$/).regex(/^[ -~]*$/);
```
```ts
const ASCII_RANGE = /^[ -~]*$/;
const PasswordSchema = z.string().regex(ASCII_RANGE).regex(/^[ -~]*$/);
```

### 💻 System info

```shell
Not relevant
```

### 🙋 Contribution

- [x] I am willing to open a pull request to fix this

**Comment by marcalexiei** [comment]
You are right, thanks for the report!

Two followups:

1. Exclude `regex`, `includes`, `startsWith`, `endsWith`, `overwrite` in `eslint-plugin-zod`.
   They have the same problem as `regex`.

2. Handle the genuinely duplicated regex in `no-conflicting-checks`.
   It already owns the "redundant" message and already knows `regex` as a check.
   Going through the shared constraint collector also covers `zod/mini`'s `z.string().check(z.regex(A), z.regex(A))` for free.

> [!NOTE]
> same identifier twice and same literal twice are in.
> `.regex(ASCII_RANGE).regex(/^[ -~]*$/)` needs const resolution, better left it out.

I'll open #436 for the second followup.

**Comment by marcalexiei** [comment]
Hey @noahbald, would you like to submit a PR for this issue?
If you don't have time I can take care of this myself!

**Comment by noahbald** [comment]
I’ll have time to submit a PR :)

**Comment by noahbald** [comment]
question: does `startsWith` and `endsWith` make sense? I presume chaining them means the validation is "and-ed", so the following would never match anything `z.startsWith('foo').startsWith('bar')`.

**Comment by marcalexiei** [comment]
They're fine to exclude: [`no-conflicting-checks`](https://github.com/marcalexiei/eslint-zod/blob/main/plugins/eslint-plugin-zod/docs/rules/no-conflicting-checks.md) already reports `z.string().startsWith('foo').startsWith('bar')` as `impossibleCase`.

`no-duplicate-schema-methods` only sees the method name, 
so it can't tell that case from `includes('a').includes('b')`.
Whatever depends on the arguments belongs in `no-conflicting-checks`.

Good catch on `z.string().startsWith('a').startsWith('abc')` though: 
redundant, not impossible, and nothing reports it once this lands.
I'll cover it in #436.
